### DISCUSSION TITLE
Multiply 서버 설정 문제) schemas를 설정해도 연결되지 않아요.

### DISCUSSION THREAD

**Discussion by YoungAhAh** [naverz/zepeto-studio-kor#1177] [5. Q&A]
안녕하세요. 멀티플레이 맵을 만드려는데 상당한 어려움을 겪고 있습니다.
처음부터 영상을 다시 따라해보고있는데요,

![image](https://user-images.githubusercontent.com/111338935/184813531-99542746-7b57-4ad2-8b55-987f47155e9d.png)
Schemas를 이렇게 Apply까지 해도

![image](https://user-images.githubusercontent.com/111338935/184813572-1916893a-99b6-4e8e-ad64-f1357afd8b4b.png)
Player값이 읽히지 않는다고 뜨고,
콘솔 로그로는 id가 제대로 뜨는게 아니라 말 그대로 '[OnJoin] sessionId: ${client.sessionId}, HashCode: ${client.hashCode}, userId:${client.userId}' >> 이 문구만 뜹니다.

어떻게 해결해야할까요? 많이 헤매고 있어서 이후에 질문이 또 생길수도 있겠습니다 ㅠ

**Comment by zlmrzlo** [comment]
visual studio로 했을 때 비슷한 문제를 겪은 적이 있습니다. 혹시 IDE가 vs Code인가요 visual studio인가요?

**Reply by YoungAhAh** [reply]
안녕하세요, 글 달아주셔서 감사합니다.
저는 VS code를 쓰고있습니다.

**Reply by YoungAhAh** [reply]
말씀해주신 것 보고 생각해보니 혹시 이게 원인일 수도 있을 것 같은데요,
![image](https://user-images.githubusercontent.com/111338935/184818919-e36cd5c4-ad14-48c4-96b4-82e8f61fffc5.png)
이상하게 ts파일들이 동영상 형식처럼 떠서 더블클릭으로는 열 수가 없더라고요.
그래서 VS code에 드래그앤드롭하는식으로 쓰고있어요. (이게 문제일수는 있으나 어떻게 해결가능한진 모르겠습니다)

**Reply by zlmrzlo** [reply]
그거 파일 경로 들어가서 해당 파일을 오른쪽 버튼 클릭으로 연결 프로그램 눌러서 vscode로 읽어올 수 있도록 하면 될 것 같아요.

**Reply by zlmrzlo** [reply]
그리고 preference에서 external tools 에서 vscode로 되어 있는지 확인 한 번 해보세요. 안되어 있다면 vscode 바꾸시고요.

**Reply by YoungAhAh** [reply]
알려주셔서 감사합니다!
일단 기본적으로 VS code로 열리는건 해결했는데 본문 문제는 계속 나타나네요.

**Reply by zlmrzlo** [reply]
아... 생각해보니 저건 player가 변수로 사용되지 않아서 생기는 줄 같아요.

**Reply by zlmrzlo** [reply]
멀티플레이 예제 따라하시면 자연스레 저 문제는 해결되지 않을까 싶습니다.

**Reply by YoungAhAh** [reply]
앗 그렇군요...그러면 
![image](https://user-images.githubusercontent.com/111338935/184820490-f5b49e5c-fbb9-4824-9b92-da48dad031a6.png)
이렇게만 뜨는데엔 또 다른 원인이 있다는거네요... 어렵군요

**Reply by zlmrzlo** [reply]
혹시 서버 스타트가 되어 있는지 확인이 필요해 보이고 아이디 로그인이 되어있는지도 확인이 필요해 보이네요

**Reply by zlmrzlo** [reply]
그리고 저게 콘솔창에 뜨는 거라면 ${} 부분에 변수가 들어간 것이 아니라 그것을 ""나 ''를 사용해서 문자열로 만들어서 저렇게 나온 것이지 않을까 싶어요

**Reply by YoungAhAh** [reply]
![image](https://user-images.githubusercontent.com/111338935/184821536-e0ecd2d4-5a8f-461e-8d5d-75b6e37e8552.png)
서버는 이런식으로 켜져있는듯하고, 로그인도 되어있습니다
아래에 가이드영상과 제 코드를 첨부했는데 뭔가 다른 점이 있을까요?
답변해주셔서 다시한번 감사합니다

**Reply by YoungAhAh** [reply]
아래에 달아주신 댓글 확인했습니다. 감사합니다 ^^

**Comment by YoungAhAh** [comment]
![image](https://user-images.githubusercontent.com/111338935/184820760-b5d27003-01e7-41a3-9994-29453090133e.png)
이게 가이드 영상에서 보이는 코드고
![image](https://user-images.githubusercontent.com/111338935/184820859-a5106df7-5db0-437c-af63-58960213b33a.png)
이게 제가 옮겨적은 내용인데
${} 가 파란색으로 안변해서 그런 것 같아요

**Reply by zlmrzlo** [reply]
tab 키 위쪽에 있는 `를 사용하시면 됩니다.
enter 키 왼쪽에 있는 '를 사용하시면 안되고요

**Reply by YoungAhAh** [reply]
아 그런거였군요! 정말 감사합니다
